### PR TITLE
Config Overhaul

### PR DESCRIPTION
**Pull request by kn0wmad**

### CODE REVIEW

File: scripts/procedures/getConfig.ts [hunk 1]
```diff
@@ -594,48 +598,45 @@ export const [getConfig, setConfigMatcher] = compat.getConfigAndMatcher({
...
+          "Plugins are subprocesses that provide extra functionality and run alongside the lightningd process inside \nthe main Core Lightning container in order to communicate directly with it.\nTheir source is maintained separately from that of Core Lightning itself.",
         spec: {
           rebalance: {
             type: "boolean",
             name: "Rebalance",
             description:
-              "Enables the `rebalance` rpc command, which moves liquidity between your channels using circular payments.\nSee `help rebalance` on the CLI or in the Spark console for usage instructions.\n\nSource: https://github.com/lightningd/plugins/tree/master/rebalance\n",
+              "Enables rebalancing, which moves liquidity between your channels using circular payments.  As of early 2024, this can only be done in the CLI.  <b>Default: False</b><br><b>Source:  https://github.com/lightningd/plugins/tree/master/rebalance</b>",
```
Comment (Dominion5254) [inline scripts/procedures/getConfig.ts:607]: To my reading, this plugin was always only usable via CLI, unless there was an client that was able to make this RPC call that is no longer supported/available.

https://github.com/lightningd/plugins/tree/master/rebalance

**Review by Dominion5254 — COMMENTED** [review]
Just a couple minor comments. I still need to cross reference with setConfig.ts to make sure there are no changes required there as a result of these changes.

File: scripts/procedures/getConfig.ts [hunk 2]
```diff
@@ -341,27 +339,24 @@ export const [getConfig, setConfigMatcher] = compat.getConfigAndMatcher({
...
-            warning:
-              "Dual funding is an experimental feature which can cause your node automatically to commit on-chain funds into channels that may not be profitable. Use at your own risk.",
             tag: {
               id: "enabled",
-              name: "Dual Funding Enabled",
+              name: "Dual Funding",
+              warning: "Dual funding is an experimental feature which can cause your node to automatically commit on-chain funds into channels that may or may not be profitable. <b>Use at your own risk!</b>",
               description:
-                "<ul><li><b>Disabled</b>: Disable Dual Funding</li><li><b>Enabled</b>: Enable Dual Funding</li></ul>",
+                "Dual Funding enables use of the channel opening protocol v2, in which both channel parties commit funds into the channel at opening. This potentially solves all sorts of problems with liquidity on the Lightning Network, but is currently experimental and only implemented by Core Lightning so far.<br>See https://blog.blockstream.com/setting-up-liquidity-ads-in-c-lightning/ for more details.  <b>Default: Disabled</b>",
```
[the remark this answers is not in the document]
Comment (kn0wmad) [reply scripts/procedures/getConfig.ts:359]: I think it comes from [CLN](https://docs.corelightning.org/docs/hooks#openchannel2) where in code this is known as openchannel2.  I'm not sure where bitcoinops/voltage got the name from, but I have no preference

File: scripts/procedures/getConfig.ts (hunk 1 above)
Comment (kn0wmad) [reply scripts/procedures/getConfig.ts:606]: Spark was a "UI," where you could enter these calls, which is now deprecated.  I'm simply making a note that this is not possible in any UI tools at this time.  Hopefully RTL will pick up the ball here

**Pull request merged**